op-challenger/fault/factory.go

```diff
@@ -56,14 +56,19 @@ func (l *gameLoader) FetchAllGamesAtBlock(ctx context.Context, blockNumber *big.
 		return nil, fmt.Errorf("failed to fetch game count: %w", err)
 	}
 
-	games := make([]FaultDisputeGame, gameCount.Uint64())
-	for i := uint64(0); i < gameCount.Uint64(); i++ {
-		game, err := l.caller.GameAtIndex(callOpts, big.NewInt(int64(i)))
+	games := make([]FaultDisputeGame, 0)
+	if gameCount.Uint64() == 0 {
+		return games, nil
+	}
+	for i := gameCount.Uint64(); i > 0; i-- {
+		game, err := l.caller.GameAtIndex(callOpts, big.NewInt(int64(i-1)))
```

Comment:
Very minor nit: I tend to prefer to do the off by one adjustment in the for loop when counting down, but you don't have to.
```suggestion
	for i := gameCount.Uint64() - 1; i >= 0; i-- {
		game, err := l.caller.GameAtIndex(callOpts, big.NewInt(int64(i)))
```

The trick of counting backwards here & relying on the sorted array is really nice.

Reply:
The problem of doing this is that in the last iteration of the loop, the `i` value will decrement from 0 and underflow, resulting in the loop continuing at the max uint64 value. I tried this out initially.

Reply:
yea, that won't work with a uint.